drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -666,9 +682,10 @@ def _mark_complete_and_publish_grade(self):
 
         # There's no going back from "completed" status to "incomplete"
         self.completed = self.completed or self._is_answer_correct() or not self.attempts_remain
-        grade = self._get_grade()
+        grade = self._calculate_grade()
         # ... and from higher grade to lower
-        if grade > self.grade:
+        current_grade = self._get_grade_if_set()
+        if current_grade is None or grade > current_grade:
```

Comment:
@nasthagiri @sstack22 keeping the highest score is by design:
* [if you have implemented already, no worries at all"](https://openedx.atlassian.net/browse/SOL-1944?focusedCommentId=205655&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-205655))
* and than [latest version implemented just keeps track of highest achieved score and only submits if the new score was higher](https://openedx.atlassian.net/browse/SOL-1944?focusedCommentId=205666&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-205666)

Than the discussion around this stopped. We might want to revisit it, but I would suggest changing that in a separate PR if necessary.